web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1143,31 +1143,65 @@ namespace com.keyman.keyboards {
      *  @param  {string}   cmd        command string
      *  @param  {boolean?} byLanguage if true, context=languages, else context=keyboards
      **/
-    keymanCloudRequest(cmd: string, byLanguage?: boolean) {
-      var URL='https://api.keyman.com/cloud/4.0/', tFlag,
-        Lscript = this.keymanweb.util._CreateElement('script');
-
-      URL = URL + ((arguments.length > 1) && byLanguage ? 'languages' : 'keyboards')
-        +'?jsonp=keyman.register&languageidtype=bcp47&version='+this.keymanweb['version'];
-
+    keymanCloudRequest(cmd: string, byLanguage?: boolean) {    
       var kbdManager = this;
+      var keymanweb = this.keymanweb;
+
+      var TIMEOUT_KEY = "cloud request timed out";
+
+      // Some basic support toward #5044, but definitely not a full solution toward it.
+      // We'd want to define an 'associative array' for registration promises, using
+      // the timer ID as the 'key' that would help us retrieve the matching Promise.
+      //
+      // Given that, if the Promise were wrapped and `resolve` and `reject` captured,
+      // the `register` function could then forward the succesfully-retrieved stubs to
+      // the `resolve` method upon success.
+      let promise = new Promise<void>(function(resolve, reject) {
+        var URL='https://api.keyman.com/cloud/4.0/';
+        var tFlag: string;
+        let Lscript: HTMLScriptElement = keymanweb.util._CreateElement('script');
+
+        URL = URL + ((arguments.length > 1) && byLanguage ? 'languages' : 'keyboards')
+          +'?jsonp=keyman.register&languageidtype=bcp47&version='+keymanweb['version'];
+  
+        // Set callback timer
+        let timeoutID = window.setTimeout(function() {
```

Comment:
```suggestion
        const timeoutID = window.setTimeout(function() {
```

Can we tidy this up further with an arrow function?

Reply:
I'm wary to do so; I seem to remember arrow functions giving us problems in Web at some point.  _Maybe_ it's resolved now that we've dropped IE.

Reply:
I thought that Typescript would transpile arrow funcs down to normal functions with `  "target": "es5",` in tsconfig.json?